src/search-modal/SearchUI.test.jsx

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable react/prop-types */
+// @ts-check
+import React from 'react';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { initializeMockApp } from '@edx/frontend-platform';
+import { AppProvider } from '@edx/frontend-platform/react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import {
+  fireEvent,
+  render,
+  waitFor,
+  getByLabelText as getByLabelTextIn,
+} from '@testing-library/react';
+import fetchMock from 'fetch-mock-jest';
+
+// @ts-ignore
+import mockResult from './__mocks__/search-result.json';
+import SearchUI from './SearchUI';
+
+// mockResult contains only a single result - this one:
+const mockResultDisplayName = 'Test HTML Block';
+
+const queryClient = new QueryClient();
+
+// Default props for <SearchUI />
+const defaults = {
+  url: 'http://mock.meilisearch.local/',
+  apiKey: 'test-key',
+  indexName: 'studio',
+  courseId: 'course-v1:org+test+123',
+};
+const searchEndpoint = 'http://mock.meilisearch.local/multi-search';
+
+/** @type {React.FC<{children:React.ReactNode}>} */
+const Wrap = ({ children }) => (
+  <AppProvider>
+    <IntlProvider locale="en" messages={{}}>
+      <QueryClientProvider client={queryClient}>
+        {children}
+      </QueryClientProvider>
+    </IntlProvider>
+  </AppProvider>
+);
+
+describe('<SearchUI />', () => {
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+    fetchMock.post(searchEndpoint, (_url, req) => {
+      const requestData = JSON.parse(req.body?.toString() ?? '');
+      const query = requestData?.queries[0]?.q ?? '';
+      // We have to replace the query (search keywords) in the mock results with the actual query,
+      // because otherwise Instantsearch will update the UI and change the query,
+      // leading to unexpected results in the test cases.
+      mockResult.results[0].query = query;
+      // And create the required '_formatted' field; not sure why it's there - seems very redundant. But it's required.
+      // eslint-disable-next-line no-underscore-dangle, no-param-reassign
+      mockResult.results[0]?.hits.forEach((hit) => { hit._formatted = { ...hit }; });
+      return mockResult;
+    });
+  });
+
+  afterEach(async () => {
+    // Unfortunately it seems a slight delay is necessary to capture all the asynchronous Meilisearch calls.
+    // If we don't do this, some network requests happen while our fetchMock is inactive, resulting in a warning.
+    // await new Promise(r => { setTimeout(r, 500); });
```

Comment:
Do we need this commented code?

Reply:
Good catch! It's not needed anymore; I found a much cleaner way to handle it :)